core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitor.scala

```diff
@@ -284,90 +290,89 @@ class AvroUpsertMonitorActor(val pollInterval: FiniteDuration,
       self ! None
   }
 
-  private def importEntities(message: PubSubMessage) = {
-    val attributes = parseMessage(message)
-
-    val importFuture = for {
-      workspace <- dataSource.inTransaction { dataAccess =>
-        dataAccess.workspaceQuery.findByName(attributes.workspace).map {
-          case Some(workspace) => workspace
-          case None =>
-            throw new RawlsException(s"Workspace ${attributes.workspace} not found while importing entities")
+  private def importEntities(message: PubSubMessage) =
+    new AvroUpsertMonitorMessageParser(message, dataSource).parse flatMap { attributes =>
```

Comment:
message parsing has changed from a synchronous call to a Future (because it may need to make a db query), so I had to add this extra flatMap into the chain.